src/initializers/kafka/base-kafka-handler.ts

```diff
@@ -0,0 +1,43 @@
+import { KafkaMessage, NConsumer } from 'sinek';
+import Kafka from './kafka';
+import { Logger } from 'log4js';
+import { flatten } from 'lodash';
+
+export default abstract class BaseKafkaHandler {
+  consumer: NConsumer;
+  topic: string;
+  batchSize: number;
+  logger: Logger;
+
+  constructor(kafka: Kafka, options: { topic: string; logger: Logger; batchSize?: number; }) {
+    const { topic, batchSize, logger } = options;
+    this.topic = topic;
+    this.batchSize = batchSize || 1;
+    this.logger = logger;
+    this.consumer = kafka.createConsumer(this.topic);
+    this.consumer.connect().then(() => this.consume().catch(err => this.logger.error(err)));
+  }
+
+  async abstract handle(message: any): Promise<any>;
```

Comment:
Can you add typings in message? Or is it only the value?

Reply:
I'll have to check. For now I need only the ok for the interface

Reply:
The example is clear. I like it

Reply:
This is only the value, `await Promise.all(flatten([messages]).map(m => this.handle(m.value)));`
Moreover, this is already an object that we can use 
```
      false,
      true, //Receive as object
```

Reply:
I believe we can close this for now. Maestro has a working example, we can come back for any improvements if the need arises.